File: C/7kyu/Character Counter.c
/*
https://www.codewars.com/kata/56786a687e9a88d1cf00005d/train/c
*/

#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool validate_word(const char *word);
int count_characters(const char *str, char c);
#define formatBool(b) ((b) ? "true" : "false")


int main(){
    //test cases
    printf("%s --> true\n", formatBool(validate_word("")));
    printf("%s --> true\n", validate_word("ABCabc")? "true":"false");
    printf("%s --> false\n", validate_word("abc:abc")? "true":"false");

    return 0;
}


bool validate_word(const char *word){

    int size = strlen(word);
    int i;
    int count = count_characters(word, toupper(word[0]));

    for (i = 1; i < size; i++){
        if (count_characters(word, toupper(word[i])) != count) return false;
    }

    return true;
}


int count_characters(const char *str, char c){

const char *p = str;
int count = 0;

do {
    if (toupper(*p) == c)count++;
    }while(*(p++));
return count;
}

File: Python/7kyu/Character Counter.py
# https://www.codewars.com/kata/56786a687e9a88d1cf00005d/train/python

def validate_word(word):
    # return True if not sum([word.upper().count(n) for n in word.upper()]) %len(word) else False
    l = []
    for c in word.upper():
        if c not in l:
            l.append(word.upper().count(c))
    print(l)
    print(sum(l)%len(l))
    return True if len(set(l)) == 1 else False


if __name__ == "__main__":
    print(validate_word("abcabc"),True)
    print(validate_word("Abcabc"),True)
    print(validate_word("AbcabcC"),False)
    print(validate_word("AbcCBa"),True)
    print(validate_word("pippi"),False)
    print(validate_word("abcabcd"),False)
    print(validate_word('a?b?xpxBabbbaPb?Xp'), False)


"""
from collections import Counter

def validate_word(word):
    return len(set(Counter(word.lower()).values())) == 1
"""

"""
def validate_word(word):
    word = word.lower()
    return len(set(word.count(c) for c in word)) == 1
"""


"""
def validate_word(word):
    word = word.lower()
    return all(word.count(i) == word.count(word[0]) for i in word)
"""